complex/ft_domain.c

```diff
@@ -192,8 +192,9 @@ static int ft_setup_xcontrol_bufs(struct ft_xcontrol *ctrl)
 		memset(ctrl->buf, 0, size);
 	}
 
-	if ((fabric_info->mode & FI_LOCAL_MR) && !ctrl->mr) {
-		ret = fi_mr_reg(domain, ctrl->buf, size, FI_RECV | FI_SEND,
+	if ((fabric_info->domain_attr->mr_mode & FI_MR_LOCAL) && !ctrl->mr) {
```

Comment:
There's a similar check in ft_setup_atomic_control that probably needs to be updated as well.

Reply:
Done.